robottelo/api/utils.py

```diff
@@ -833,7 +833,7 @@ def update_provisioning_template(name=None, old=None, new=None):
     """
     temp = (
         entities.ProvisioningTemplate()
-        .search(query={'per_page': 1000, 'search': f'name="{name}"'})[0]
+        .search(query={'per_page': '1000', 'search': f'name="{name}"'})[0]
```

Comment:
Does "per_page" is required?

Reply:
@devendra104 Yes, very much since by default it searches in first 20 templates and then returns None. We have to either iterate over pages or use the higher number like 1000. 

Even though its 1000 the actual number of templates should be around 180, but it might get higher if any test adds the templates manually.

Reply:
Thanks, @jyejare for clarifying it, I think to fix the entries in per_page is not a good approach because count varies from time to time, if we use a loop then we can avoid the per_page related failure in the future.

Reply:
@devendra104 I know running through the loop would be more dynamic but the 1000 count is such high. I am sure that we ll never exceed that limit. :)

Also, lookup operation from Satellite API in 1000 entities is faster than nailgun looping search. I choose the faster way :)